### ISSUE TITLE
[Feature Request] MSAL C++ library

### ISSUE THREAD

**Issue by parthasarathi204** [AzureAD/microsoft-authentication-library-for-dotnet#3744]
**Is your feature request related to a problem? Please describe.**
Yes. Like to use WAM broker using msal CPP code. We can do writing an interops library. But as you have the cPP msal library already , can you please make this publicly available.  Referenace as
'New WAM Preview in MSAL 4.44+
The New MSAL WAM Preview is an abstraction layer based on MSAL C++ which fixes a number of issues with the existing WAM implementation. New applications should use this implementation."

**Describe the solution you'd like**
Prefer to use MSAL C++ library . Will be easier to cpp application written in ADAL

**Describe alternatives you've considered**
Wrote an interops dll based on the dotnet MSAL library.

**Additional context**
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/wam

**Comment by bgavrilMS** [comment]
Yes, there are plans to open source MSAL C++, but am not sure of the deadline. 

An alternative to writing your own interop code is to use .NET's AOT compiler https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/

**Comment by parthasarathi204** [comment]
Can someone please specify the deadline? CC @iulico-1

**Comment by iulico-1** [comment]
Making MSAL.CPP public is in our long-term plans but there's no concrete timeline we can commit to at this point.

**Comment by bgavrilMS** [comment]
Closing this as it is not related to MSAL .NET per se.

**Comment by bgavrilMS** [comment]
I am not aware of updates on this @whgibbo. Best course of action would be to use a standard OAuth / OIDC library, since AAD is standards complaint. Or call into MSAL .NET ...
